Integrations/src/main/java/io/deephaven/integrations/learn/Gatherer.java

```diff
@@ -1,30 +1,55 @@
 package io.deephaven.integrations.learn;
 
 import io.deephaven.db.v2.sources.ColumnSource;
+// REMOVE THIS AFTER REMOVING makeIndexSet!
+import io.deephaven.db.tables.Table;
 
+/**
+ * Gatherer takes Deephaven columnar data and places it into a buffer to be used by Python.  The Python object will take
+ * data from the buffer and use it to construct a 2d array of specified size.
+ */
 public class Gatherer {
 
+    /**
+     * Generate an index set from a table.
+     * THIS IS JUST FOR EASY PYTHON TESTING.  REMOVE BEFORE PROD.
+     *
+     * @param t table
+     * @return index set
+     */
+    public static IndexSet makeIndexSet(final Table t) {
+        final IndexSet is = new IndexSet(t.intSize());
+
+        for(long idx : t.getIndex()) {
+            is.add(idx);
+        }
+
+        return is;
+    }
+
     /**
      * Copy data from a table into a 2d tensor of Booleans.
      *
      * @param indexSet indices of the rows of the table to put into the tensor
      * @param columnSources columns of data to put into the tensor
+     * @param transpose a boolean indicating the user's intent to return a transposed Python object
      * @return contiguous RAM allocated for the tensor. When a numpy tensor is passed in for this argument, jpy will
      *         handle passing the memory reference as a 1d java array here.
      */
-
-    public static Boolean[] booleanTensorBuffer2D(final IndexSet indexSet, final ColumnSource<?>[] columnSources) {
+    public static boolean[] tensorBuffer2Dboolean(final IndexSet indexSet,
+                                                  final ColumnSource<?>[] columnSources,
+                                                  final boolean transpose) {
 
         final int nRows = indexSet.getSize();
         final int nCols = columnSources.length;
-        final Boolean[] tensor = new Boolean[nRows * nCols];
+        final boolean[] tensor = new boolean[nRows * nCols];
 
         int j = 0;
-        for (ColumnSource<?> cs : columnSources) {
+        for (ColumnSource<?> cs: columnSources) {
             int i = 0;
 
             for (long idx : indexSet) {
-                final int ij = nCols * i + j;
+                final int ij = transpose ? nRows * j + i : nCols * i + j;
```

Comment:
Have you printed some output to verify that this does create the transposed tensor?

Reply:
I have run tests in Python to verify that the transpose works properly.  It does.  I did, however, just run some interesting tests that didn't give the results I expected.  These results support not even bothering with this, and dealing with it only in Python.

NumPy arrays have a built in transpose:

```
a = np.array([[1, 2, 3], [4, 5, 6]])
a_transpose = a.T
```

Using the *.T method to transpose is showing slightly faster performance, and that's without setting the transpose flag in Java, which means it's using the base `ij` index.